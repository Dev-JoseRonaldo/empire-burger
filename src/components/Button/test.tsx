import { render, screen } from '@testing-library/react'

import { Button } from '.'

const { getByRole } = screen

describe('<Button />', () => {
  it('should render by primary button', () => {
    render(<Button variant="primary">Children</Button>)

    const component = getByRole('button', { name: 'Children' })

    expect(component).toHaveClass(
      'bg-primary text-white-900 px-5 py-[0.625rem] font-bold'
    )
  })

  it('should render by secondary button', () => {
    render(<Button variant="secondary">Children</Button>)

    const component = getByRole('button', { name: 'Children' })

    expect(component).toHaveClass(
      'bg-white-500 text-primary px-4 py-2 font-black'
    )
  })

  it('should render with custom class', () => {
    render(<Button className="uppercase">Children</Button>)

    const component = getByRole('button', { name: 'Children' })

    expect(component).toHaveClass('uppercase')
  })

  it('should match snapshot', () => {
    const { container } = render(<Button>Children</Button>)

    expect(container).toMatchSnapshot()
  })
})
