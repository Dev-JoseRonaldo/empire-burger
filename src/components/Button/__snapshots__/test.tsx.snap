// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should match snapshot 1`] = `
<div>
  <button
    class="flex items-center justify-center rounded-10 bg-primary font-bold text-secondary transition hover:text-primary hover:bg-transparent border-2 border-primary transition-color w-[16.188rem] text-2xl h-[2.875rem] "
  >
    Children
  </button>
</div>
`;
